gitbook-docs/streaming_api.md

```diff
@@ -50,3 +50,31 @@ The server MAY provide:
 `name`, `self` and `roles` MUST return the same values as provided in the `swname`, `self` and `roles` properties within the [DNS-SD advertisement](connection.md) (if implemented).
 
 `version` MUST be the same value as `version` within the associated endpoints list provided by the http `GET` request to `/signalk` within the [REST API](rest_api.md) (if implemented).
+
+## History playback
+
+The server MAY support history playback from a certain point in time with a specified rate.
+
+To create a WebSocket connection that plays back data the client uses the request parameter `startTime` to specify the start timestamp and the optional request parameter `playbackRate` to specify the rate. Rate value parameter is a floating point value with value `1` equal to real time playback and for example `0.5` to half the real time rate and `5` to five times real time rate. Omitting the `playbackRate` will result in real time playback.
+
+Example url for history playback streaming: `wss://localhost:3443/signalk/v1/stream?subscribe=self&startTime=2018-08-24T15:19:09Z&playbackRate=5`.
+
+The hello message for a history playback stream MUST NOT contain the `timestamp` property and MUST include the properties `startTime` and `playbackRate`. The delta stream format for history playback is the normal streaming format except for the . Timestamps will indicate the time data was originally captured.
+
```

Comment:
I think we should add: `When the server has finished playing back available data, the connection is closed.`

Reply:
Why? When is the available data exhausted? Like if I sail some, stop for lunch x minutes and turn instruments off, then sail some should the playback stop?

Reply:
What is the end and what is just a hole? Ihave data over several years with different size holes. 

We can leave this for the server to decide of course. One way is just to keep the connection hanging and skip this.